src/EditorFeatures/Core/Implementation/AddImports/AddImportsPasteCommandHandler.cs

```diff
@@ -0,0 +1,120 @@
+            if (!caretPosition.HasValue)
+            {
+                return;
+            }
+
+            nextCommandHandler();
+
+            if (!args.SubjectBuffer.CanApplyChangeDocumentToWorkspace())
+            {
+                return;
+            }
+
+            // Don't perform work if we're inside the interactive window
+            if (args.TextView.IsBufferInInteractiveWindow(args.SubjectBuffer))
+            {
+                return;
+            }
+
+            if (!args.SubjectBuffer.GetFeatureOnOffOption(FeatureOnOffOptions.AddImportsOnPaste))
+            {
+                return;
+            }
+
+            // Create a tracking span from the pre-paste caret position that will grow as text is inserted.
+            var trackingSpan = caretPosition.Value.Snapshot.CreateTrackingSpan(caretPosition.Value.Position, 0, SpanTrackingMode.EdgeInclusive);
+
+            // Applying the post-paste snapshot to the tracking span gives us the span of pasted text.
+            var snapshotSpan = trackingSpan.GetSpan(args.SubjectBuffer.CurrentSnapshot);
+            var textSpan = TextSpan.FromBounds(snapshotSpan.Start, snapshotSpan.End);
+
+            AddMissingImportsForPaste(args, executionContext, textSpan);
+        }
+
+        public CommandState GetCommandState(PasteCommandArgs args, Func<CommandState> nextCommandHandler)
+            => nextCommandHandler();
+
+        private void AddMissingImportsForPaste(PasteCommandArgs args, CommandExecutionContext executionContext, TextSpan textSpan)
+        {
+            var sourceTextContainer = args.SubjectBuffer.AsTextContainer();
+            if (!Workspace.TryGetWorkspace(sourceTextContainer, out var workspace))
+            {
+                return;
+            }
+
+            var document = sourceTextContainer.GetOpenDocumentInCurrentContext();
+            if (document is null)
+            {
+                return;
+            }
+
+            using var _ = executionContext.OperationContext.AddScope(allowCancellation: true, EditorFeaturesResources.Adding_missing_import_directives);
+            var cancellationToken = executionContext.OperationContext.UserCancellationToken;
+
+#pragma warning disable VSTHRD102 // Implement internal logic asynchronously
+            var updatedDocument = _threadingContext.JoinableTaskFactory.Run(() => AddMissingImportsAsync(document, textSpan, cancellationToken));
+#pragma warning restore VSTHRD102 // Implement internal logic asynchronously
+            if (updatedDocument is null)
+            {
+                return;
+            }
+
+            updatedDocument.Project.Solution.Workspace.TryApplyChanges(updatedDocument.Project.Solution);
+        }
+
+        private static async Task<Document?> AddMissingImportsAsync(Document document, TextSpan textSpan, CancellationToken cancellationToken)
+        {
+            var addMissingImportsService = document.GetRequiredLanguageService<IAddMissingImportsFeatureService>();
```

Comment:
you exported the command handler for all roslyn languages.  however, this interface won't exist for TS/Xaml/F#.  so this will crash.

Reply:
o.o. Time to add an integration test for TS/Xaml/F# paste? I assumed it existed...

Reply:
I changed the content type to CSharp/VB AND allowed this to be null. I don't think we should error here if this is missing, we can no-op instead.